src/codeu/chat/server/Server.java

```diff
@@ -58,20 +61,57 @@ public Server(Uuid id, byte[] secret, Relay relay) {
 
     this.controller = new Controller(id, model);
     this.relay = relay;
-  }
 
-  public void syncWithRelay(int maxReadSize) throws Exception {
-    for (final Relay.Bundle bundle :  relay.read(id, secret, lastSeen, maxReadSize)) {
-      onBundle(bundle);
-      lastSeen = bundle.id();
-    }
+    timeline.scheduleNow(new Runnable() {
+      @Override
+      public void run() {
+        try {
+
+          LOG.info("Reading update from relay...");
+
+          for (final Relay.Bundle bundle : relay.read(id, secret, lastSeen, 32)) {
```

Comment:
As per discussions below - this will later change to use the hint sent back by the relay. For now I will update it to use a documented constant and put in a TODO.